src/views/login/components/login.js

```diff
@@ -4,16 +4,38 @@ import './login.scss';
 export default class Login extends Component {
 
     static propTypes = {
+        usernameChanged: PropTypes.func.isRequired,
+        passwordChanged: PropTypes.func.isRequired,
+        login: PropTypes.func.isRequired,
+
+        usernameField: PropTypes.string.isRequired,
+        passwordField: PropTypes.string.isRequired,
+        history: PropTypes.object.isRequired,
     };
 
+    handleUsernameChanged = (e) => {
+        const { usernameChanged } = this.props;
+        usernameChanged(e.target.value);
+    }
+
+    handlePasswordChanged = (e) => {
+        const { passwordChanged } = this.props;
+        passwordChanged(e.target.value);
+    }
+
+    handleLoginClicked = () => {
+        const { login, usernameField, passwordField, history } = this.props;
+        login(usernameField, passwordField, history);
```

Comment:
in `login-connector.js` i create the login component and bind the user action creators with `{...bindActionCreators(userActionCreators, dispatch)}`. so this gives us access to the user action creator functions inside the login component as props